btree/btree.c

```diff
@@ -2039,36 +2075,146 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	node_fix(tgt, tx);
 }
 
+/**
+ * calc_shift is used to calculate the shift value from the given size.
+ * Shift is the exponent of nearest power-of-2 value greater than or equal to
+ * size. 
+ * 
+ * @param size It is the number of bytes
+ * @return int returns the shift value.
+ */
+
+int calc_shift(int size)
+{
+	unsigned int sample = (unsigned int) size;
+	unsigned int pow = 0;
+
+	while (sample > 0)
+	{
+		sample >>=1;
+		pow += 1;
+	}
+
+	return pow - 1;
+}
+
+/**
+ * btree_create_tick function is the main function used to create btree.
+ * It traverses through multiple states to perform its operation.
+ * 
+ * @param smop Represents the state machine operation
+ * @return int64_t It returns the next state to be executed.
+ */
+
+int64_t btree_create_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op *bop		= M0_AMB(bop,smop,bo_op);
+	struct m0_btree_oimpl *oi	= bop->bo_i;
+	struct m0_btree *curr_btree 	= bop->bo_arbor;
+	struct td *tree;
+	struct m0_btree_idata *data;
+	struct segaddr curr_addr;
+
+	switch(bop->bo_op.o_sm.sm_state) 
+	{
+		case P_INIT:
+			data = &bop->b_data;
+
+			oi = m0_alloc(sizeof(struct m0_btree_oimpl));
+			bop->bo_i = oi;
+			curr_btree = m0_alloc(sizeof(struct m0_btree));
+			bop->bo_arbor = curr_btree;
```

Comment:
we can directly store pointer in bop->bo_arbor, no need for curr_btree

Reply:
Thanks.
Updated in latest commit i.e  d118823.